homeassistant/components/synology_dsm/camera.py

```diff
@@ -100,7 +102,11 @@ async def stream_source(self) -> str:
         """Return the source of the stream."""
         if not self.available:
             return None
-        return self._camera.live_view.rtsp
+
+        url = urlparse(self._camera.live_view.rtsp)
+        return url._replace(
+            netloc=self._hostname.join(url.netloc.rsplit(url.hostname, 1))
+        ).geturl()
```

Comment:
I guess this is more fitting logic to go into the upstream synology_dsm Python package?

Reply:
Not sure about that. I have this a thought originally but I don't see why synology_dsm would touch data returned by Synology. Seems to be the role of the integrator.

Reply:
I don't agree. Home Assistant should consume data provided by a library and not contain modify the data provided by it.

Reply:
Added upstream for consideration (https://github.com/hacf-fr/synologydsm-api/pull/9).

Reply:
@Quentame ☝️ if you could take a look please.